src/passport-saml/saml.ts

```diff
@@ -8,18 +8,27 @@ import * as xmldom from 'xmldom';
 import * as url from 'url';
 import * as querystring from 'querystring';
 import * as xmlbuilder from 'xmlbuilder';
+// @ts-expect-error no typings for xml-encryption
 import * as xmlenc from 'xml-encryption';
 import * as util from 'util';
 import {CacheProvider as InMemoryCacheProvider} from './inmemory-cache-provider';
 import * as algorithms from './algorithms';
 import { signAuthnRequestPost } from './saml-post-signing';
 import type { Request } from 'express';
+import { ParsedQs } from 'qs';
+import { AudienceRestrictionXML, AuthenticateOptions, AuthorizeOptions, AuthorizeRequestXML, CertCallback, LogoutRequestXML, Profile, RequestWithUser, SamlScopingConfig, ServiceMetadataXML, XMLInput, XMLObject, XMLOutput, XMLValue } from './types';
```

Comment:
Is it a mistake that that `import` is called on the `types` file twice?

I'm working on merging this branch and plan to simplify this.